libs/agno/agno/workflow/step.py

```diff
@@ -261,10 +261,15 @@ def execute(
                                 run_context,
                             ):  # type: ignore
                                 if isinstance(chunk, (BaseRunOutputEvent)):
-                                    if isinstance(chunk, (RunContentEvent, TeamRunContentEvent)):
-                                        content += chunk.content if chunk.content is not None else ""
-                                    elif isinstance(chunk, (RunCompletedEvent, TeamRunCompletedEvent)):
-                                        content = chunk.content if chunk.content is not None else ""
+                                    if isinstance(chunk, (RunContentEvent, TeamRunContentEvent)) and chunk.content is not None:
+                                        # Its a regular chunk of content
+                                        if isinstance(chunk.content, str):
+                                            content += chunk.content
+                                        # Its the BaseModel object, convert it to a dictionary
+                                        elif isinstance(chunk.content, BaseModel):
+                                            content = chunk.content.model_dump()
```

Comment:
When is RunContentEvent.content a BaseModel? and do we want to overwrite the content var here like that?